assets/src/edit-story/components/draganddrop/index.js

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import { useUploader } from '../../app/uploader';
+import { ReactComponent as UploadIcon } from './icons/upload.svg';
+
+const DragandDropComponent = styled.div`
+  min-width: 100%;
+  min-height: 100%;
+`;
+const DragandDropOverContent = styled.div``;
+
+const DragandDropOverLayWrapper = styled.div`
+  width: 100%;
+  height: 100%;
+  background-color: rgba(0, 0, 0, 0.6);
+  position: absolute;
+  top: 56px;
+  left: 0;
+  z-index: 999;
+`;
+const Heading = styled.h4`
+  color: ${({ theme }) => theme.colors.fg.v1};
+  margin: 0;
+`;
+const Text = styled.p`
+  color: ${({ theme }) => theme.colors.fg.v1};
+`;
+
+const DragandDropOverLay = styled.div`
+  position: absolute;
+  top: 45%;
+  text-align: center;
+  width: 100%;
+`;
+
+function DragandDrop({ children }) {
+  const [isDragging, setIsDragging] = useState(false);
+  const { uploadFile } = useUploader();
+
+  const disableDefaults = (evt) => {
+    evt.preventDefault();
+    evt.stopPropagation();
+  };
+
+  const onDragEnter = (evt) => {
+    disableDefaults(evt);
+    setIsDragging(true);
+  };
+
+  const onDropHandler = (evt) => {
+    disableDefaults(evt);
+    const dt = evt.dataTransfer;
+    let files = dt.files;
+    files = [...files];
+    files.forEach(uploadFile);
+    setIsDragging(false);
+  };
+
+  return (
+    <DragandDropComponent
+      onDragStart={disableDefaults}
+      onDragOver={disableDefaults}
+      onDragLeave={disableDefaults}
+      onDragEnter={onDragEnter}
+      onDrop={onDropHandler}
+    >
+      {isDragging && (
+        <DragandDropOverLayWrapper>
+          <DragandDropOverLay>
+            <UploadIcon />
+            <Heading>{__('Upload to media library', 'web-stories')}</Heading>
+            <Text>
+              {__(
+                'You can upload jpg, jpeg, png, svg, gif and webp.',
```

Comment:
Perhaps at some point we could use something like `wp_sprintf_l()` in JS for strings like this.

Reply:
@swissspidy By default, webp and svg will not be supported here. Should it be remove from this message?

Reply:
We should only print what we can support at this point. Everything else we should file in an issue and assign to @pbakaus :)

Reply:
We should make this string dynamic depending on the list of allowed mime types from the ConfigProvider. Might need to reword it a little bit to make that easier. For example:

```
The following file types are supported:
%s
```

which then will be turned into:

```
The following file types are supported:
png, jpg, jpeg, gif, xyz, abc
```

Reply:
@swissspidy What do you think of this - https://github.com/google/web-stories-wp/pull/175/commits/b4327cdd620561e12ef080b1a6bbf4d77da2a1e7

Reply:
Thanks! 👍 

I'd just optimize the PHP part a little bit:

```php
/**
 * Returns a flat list of allowed file types (e.g. jpeg, gif, png, mp4).
 *
 * @return array List of allowed file types.
 */
protected static function get_allowed_file_types() {
	$allowed_mime_types = self::get_allowed_mime_types();
	$mime_types         = [];

	foreach ( $allowed_mime_types as $key => $types ) {
		array_push( $mime_types, ...$types );
	}

	$allowed_file_types = [];
	$all_mime_types       = wp_get_mime_types();

	foreach ( $all_mime_types as $key => $type ) {
		if ( in_array( $type, $mime_types, true ) ) {
			array_push( $allowed_file_types, ...explode( '|', $key ) );
		}
	}

	return $allowed_file_types;
}
```

* Make variable names more clear, fix typos
* Avoid `array_merge` in loops

Reply:
Fixed in https://github.com/google/web-stories-wp/pull/175/commits/53eb38484c2578bb14635e25862cb5f1aac17944